### ISSUE TITLE
5.1 feature list

### ISSUE THREAD

**Issue by gregglind** [mozilla/shield-studies-addon-utils#177]
- prefs
- pioneer
- better integrated logging

**Comment by gregglind** [comment]
Add / merge Pioneer utils.

```
    // const { PioneerUtils } = require("pioneer-utils/PioneerUtils.jsm");
    // const pioneerUtilsBootstrap = require("./pioneerUtilsBootstrap.js");
```

**Comment by gregglind** [comment]
DataPermissionsChange stuff?

```
        // https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/events.html
        /* Fires whenever any 'dataPermission' changes, with the new dataPermission object.  Allows watching for shield or pioneer revocation. */
        // onDataPermissionsChange: new EventManager(
        //  context,
        //  "study:onDataPermissionsChange",
        //  fire => {
        //    const listener = (eventReference, updatedPermissions) => {
        //      fire.async(updatedPermissions);
        //    };
        //    studyApiEventEmitter.on("dataPermissionsChange", listener);
        //    return () => {
        //      studyApiEventEmitter.off("dataPermissionsChange", listener);
        //    };
        //  },
        // ).api(),
```

**Comment by motin** [comment]
Closing since this feature list has played out it's part. 
Follow up:
> prefs
Not yet implemented nor planned.
> pioneer
PR ready. Will likely be released in 5.2
> better integrated logging
This was released in 5.1.0
